static/app/components/events/interfaces/spans/traceView.tsx

```diff
@@ -23,16 +23,30 @@ type Props = {
 
 function TraceView(props: Props) {
   const traceViewRef = createRef<HTMLDivElement>();
+  const traceViewHeaderRef = createRef<HTMLDivElement>();
   const virtualScrollBarContainerRef = createRef<HTMLDivElement>();
   const minimapInteractiveRef = createRef<HTMLDivElement>();
 
+  const [isMounted, setIsMounted] = useState(false);
+
+  // Since this component is memoized, we need this hook here.
+  // renderHeader performs some expensive calculations and so we only want to render once, hence why we memoize.
+  // However, the virtualScrollbar will not be visible unless we have this effect here. This is a bit of a hack that will
+  // cause the component to rerender by setting the isMounted state, so we will re-render only once the scrollbar ref is present.
```

Comment:
memoization is only likely required here because there is a new `waterfallModel` reference... which probably doesn't need to happen if that model is identical, it's probably just a byproduct of implementation.

Reply:
But it's also because we're calling `waterfallModel.getWaterfall()` within the render function, which can be really expensive